interfaces/builtin/udisks2.go

```diff
@@ -0,0 +1,920 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const udisks2PermanentSlotAppArmor = `
+# Description: Allow operating as the udisks2. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.UDisks2",
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UDisks2{,/**}
+    interface=org.freedesktop.DBus**
+    peer=(label=unconfined),
+
+# Needed for mount/unmount operations
+capability sys_admin,
+
+# Allow reading existing mount points
+@{PROC}/@{pid}/mountinfo r,
+@{PROC}/swaps r,
+
+# Udisks checks for existing configured mount locations here
+/etc/fstab r,
+
+# Allow scanning of devices
+network netlink raw,
+/run/udev/data/b[0-9]*:[0-9]* r,
+/sys/devices/**/block/** r,
+
+# Mount points could be in /run/media/<user>/* or /media/<user>/*
+/run/systemd/seats/* r,
+/{,run/}media rw,
+/{,run/}media/** rw,
+mount /dev/{sd*,mmcblk*} -> /{,run/}media/**,
+umount /{,run/}media/**,
+
+# These should probably be patched to use $SNAP_DATA/run/...
+/run/udisks2/ rw,
+/run/udisks2/** rw,
+
+# udisksd execs mount/umount to do the actual operations
+/bin/mount ixr,
+/bin/umount ixr,
+
+# mount/umount (via libmount) track some mount info in these files
+/run/mount/utab wrl,
+/run/mount/utab.* wrl,
+
+/dev/sd* r,
+/dev/mmcblk* r,
+`
+
+var udisks2ConnectedSlotAppArmor = []byte(`
+# Allow connected clients to interact with the service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/UDisks2
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UDisks2
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# UDisks2 Manager Interface
+dbus (receive)
+    bus=system
+    path=/org/freedesktop/UDisks2/Manager
+    interface=org.freedesktop.UDisks2.Manager
+    member=LoopSetup
+    peer=(label=###PLUG_SECURITY_TAGS###),
+    
+dbus (receive)
+    bus=system
+    path=/org/freedesktop/UDisks2/Manager
+    interface=org.freedesktop.UDisks2.Manager
+    member=MDRaidCreate
+    peer=(label=###PLUG_SECURITY_TAGS###),
```

Comment:
It's fine to list our the dbus API as members if you want to only support that portion of the API, but if you want to support the whole API, perhaps drop the 'member' from these rules and condense down into:
```
dbus (receive)
    bus=system
    path=/org/freedesktop/UDisks2/Manager
    interface=org.freedesktop.UDisks2.Manager
    peer=(label=###PLUG_SECURITY_TAGS###),
```